### ISSUE TITLE
soc training loss is always nan

### ISSUE THREAD

**Issue by vishalnayak108** [ZHKKKe/MODNet#117]
![soc_loss](https://user-images.githubusercontent.com/35347836/123206356-2b0a0200-d4d9-11eb-99e7-a044c199ec33.JPG)

**Comment by kashyappiyush1998** [comment]
@vishalnayak108  and all the previous questioners. 
There seems to be a problem that image semantic segmentation is not working perfectly, hence creating a huge difference in the so-called expected near-perfect mask which is needed to make a perfect mask in soc training. 
When in soc training loop detail loss once found nan will result in semnatic_loss + detail_loss will also be nan, hence computing loss.backward() on nan loss will diverge/break our training process. 

A simple solution is to skip it. 

```
loss = soc_semantic_loss + soc_detail_loss
if(torch.isnan(loss)==False):
    loss.backward()
    optimizer.step()
else:
    print("Value has become nan")
    soc_semantic_loss = 0.0
    soc_detail_loss = 0.0
return soc_semantic_loss, soc_detail_loss

```


This code in soc_adaptation_iter, will skip the gradient computation part on the batch of images that has even a single image resulting in nan detail loss.

**Comment by vishalnayak108** [comment]
Thanks for your suggestion. I will try this out. But what if define training batch=1.Also why we are considering soc_semantic_loss = 0.0 soc_detail_loss = 0.0 for nan value ?

**Comment by kashyappiyush1998** [comment]
@vishalnayak108 this would work with training batch 1 as well, and we are using training losses 0 just for printing or logging, no general purpose

**Comment by ZHKKKe** [comment]
@kashyappiyush1998 
Thanks for your sharing. I also tested it. 
For some images, the SOC loss will be very large and let everything to be NaN. 
Skip these bad loss values is a possible solution.
